pymc3/tests/test_shape_handling.py

```diff
@@ -219,3 +226,234 @@ def test_sample_generate_values(fixture_model, fixture_sizes):
         prior = pm.sample_prior_predictive(samples=fixture_sizes)
         for rv in RVs:
             assert prior[rv.name].shape == size + tuple(rv.distribution.shape)
+
+
+class TestShapeDimsSize:
```

Comment:
Since most of the shape issues occur with multivariate variables, should we add a similar test using `MvNormal`?

Reply:
I find these parametrized tests a little hard to read. The `MvNormal`-related weirdness should be covered by `test_mvnormal_shape_size_difference` further below already, don't you think?

Reply:
They are definitely hard to read, but it seems this has two advantages over the `mvtest` below:
1. It uses the `non-dist` methods
2. It tests dims as well

Reply:
Let's defer this to another PR.
Also we'll probably cover this in the examples/update guide too.